cmd/clusterctl/client/config/reader_viper_test.go

```diff
@@ -26,28 +26,54 @@ import (
 )
 
 func Test_viperReader_Init(t *testing.T) {
-
 	g := NewWithT(t)
+
+	// Change HOME dir and do not specify config file
+	// (.cluster-api/clusterctl) in it.
+	origHomeDir := os.Getenv("HOME")
+	clusterctlHomeDir, err := ioutil.TempDir("", "clusterctl-default")
+	g.Expect(err).NotTo(HaveOccurred())
+	defer os.RemoveAll(clusterctlHomeDir)
+	os.Setenv("HOME", clusterctlHomeDir)
+	defer func() {
+		os.Setenv("HOME", origHomeDir)
+	}()
```

Comment:
I wonder if we should be actually doing this, it could have some side-effects that we don't know about

Reply:
agree. even a global var that can be overridden/restored later would be nicer:
```go
// reader_viper.go
var homeDirFunc = homedir.HomeDir
...
    viper.AddConfigPath(filepath.Join(homeDirFunc(), ...
```

```go
// reader_viper_test.go
oem := homeDirFunc
defer func() { homeDirFunc = oem }()  // restore original global var
homeDirFunc = someTestFixture
.. // run test case
```
of course, globals like this smell too. maybe instead of a global, define a homeDirFunc in the `viperReader` struct with a default initialized value of `homedir.HomeDir` and override that in the test case.

Reply:
Yeah..I was trying to find a way to write a test for this that didn't require me to make huge api changes all over the place. I think we can do the options pattern here. Thanks for the suggestions!